### PR TITLE
[Release 2.0] fix10289: correctly generate tsconfig.json with --lib

### LINKED ISSUE #10289: tsc --init with --lib produces broken tsconfig.json
Step 1:

```shell
tsc --init --lib es5,es2015
```

Produces:

```json
{
    "compilerOptions": {
        "lib": [
            "lib.es5.d.ts",
            "lib.es2015.d.ts"
        ],
        "module": "commonjs",
        "target": "es5",
        "noImplicitAny": false,
        "sourceMap": false
    },
    "exclude": [
        "node_modules"
    ]
}
```

which is broken because those are not valid values for `lib`.

This *should* produce

```json
{
    "compilerOptions": {
        "lib": [
            "es5",
            "es2015"
        ],
        "module": "commonjs",
        "target": "es5",
        "noImplicitAny": false,
        "sourceMap": false
    },
    "exclude": [
        "node_modules"
    ]
}
```

### PR DESCRIPTION
**Pull request by yuit**
Fix #10289

### CODE REVIEW

File: src/compiler/commandLineParser.ts [hunk 1]
```diff
@@ -665,6 +673,102 @@ namespace ts {
...
+            else if (optionDefinition.type === "list") {
+                return getCustomTypeMapOfCommandLineOption((<CommandLineOptionOfListType>optionDefinition).element);
+            }
+            else {
+                return (<CommandLineOptionOfCustomType>optionDefinition).type;
+            }
+        }
+
+        function getKeyOfCompilerOptionValue(value: CompilerOptionsValue, customTypeMap: Map<string | number>): string | undefined {
+            // There is a typeMap associated with this command-line option so use it  to map value back to its name
```
Comment (DanielRosenwasser) [inline src/compiler/commandLineParser.ts]: stray space after "use it"

File: src/compiler/commandLineParser.ts [hunk 2]
```diff
@@ -665,6 +673,102 @@ namespace ts {
...
+            for (const key in customTypeMap) {
+                if (hasProperty(customTypeMap, key)) {
+                    if (customTypeMap[key] === value) {
```
Comment (DanielRosenwasser) [inline src/compiler/commandLineParser.ts]: Don't we have a `forEachKey`?

Comment (yuit) [inline src/compiler/commandLineParser.ts]: I should update this map usage as @ahejlsberg PR is in

**Comment by yuit** [comment]
The failure is due to the fact that "release-2.0" has dependency on TypeScript nightly which has Map/MapLike optimization

**Comment by yuit** [comment]
I am going to close this one and open a new one for master

**Pull request closed without merging**